### PR TITLE
JIRA: AOAF-474

### PR DESCRIPTION
**Pull request by ch-skathirvel**
Description : Added new page to enter business name

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/routers/enter.business.name.router.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// If the user presses the back button, it will render the page as it apeared previously using
+// cached HTML even though the details could have already been submitted and session cleared.
+router.use(noCacheMiddleware);
+
+router.get("/", handleExceptions( async (req: Request, res: Response, _next: NextFunction) => {
+    const handler = new EnterBusinessNameHandler();
+    const { templatePath, viewData } = handler.executeGet(req, res);
+    const paDetails = getPresenterAccountDetails(req);
+    if (paDetails === undefined || paDetails.isBusinessRegistered === undefined) {
+        throw new Error('Is registered as a business question not answered, unable to enter business name');
```
[the remark this answers is not in the document]
Comment (ch-skathirvel) [reply src/routers/enter.business.name.router.ts:20]: Will change it

File: src/routers/enter.business.name.router.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+// Prevent caching on this page.
+// If the user presses the back button, it will render the page as it apeared previously using
+// cached HTML even though the details could have already been submitted and session cleared.
+router.use(noCacheMiddleware);
+
+router.get("/", handleExceptions( async (req: Request, res: Response, _next: NextFunction) => {
+    const handler = new EnterBusinessNameHandler();
+    const { templatePath, viewData } = handler.executeGet(req, res);
+    const paDetails = getPresenterAccountDetails(req);
```
[the remark this answers is not in the document]
Comment (ch-skathirvel) [reply src/routers/enter.business.name.router.ts:18]: Does that really matters? I thought to use as paDetails as a short form for presenteraccount.

File: src/routers/handlers/enter_business_name/index.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        if (isLang(lang)){
+            details.lang = lang;
+        } else {
+            throw new Error(`Invalid language: ${lang} variable should be either 'en' or 'cy'`);
+        }
+        if (!details.isBusinessRegistered){
+            // Set the business name in session, if business is not registered
+            details.businessName = req.body.businessName;
+        } else {
+            // TODO
```
[the remark this answers is not in the document]
Comment (ch-skathirvel) [reply src/routers/handlers/enter_business_name/index.ts:61]: Will remove it

File: src/routers/handlers/is_business_registered/index.ts [hunk 4]
```diff
@@ -62,13 +62,19 @@ export class IsBusinessRegisteredHandler extends GenericHandler<IsBusinessRegist
         }
         logger.debug(`is business registered with companies house: ${JSON.stringify(isRegistered)}`);
         const detail = { ...getPresenterAccountDetails(req), isBusinessRegistered };
-        setPresenterAccountDetails(req, detail);
+        let redirectUrl: string;
 
         if (isBusinessRegistered) {
-            return { redirect: "company-search" };
+            // Reset the business name value in session, when the business is registered
+            detail.businessName = null;
```
[the remark this answers is not in the document]
Comment (ch-skathirvel) [reply src/routers/handlers/is_business_registered/index.ts:69]: When we click the back from 'Enter business name' screen and if 'Yes' selected in isBusinessRegistered then clearing the business name from session

File: src/utils/session.ts [hunk 5]
```diff
@@ -39,6 +40,7 @@ export function populatePresenterAccountDetails(req: Request): PresenterSessionD
     };
     const presenterAccountDetails = getPresenterAccountDetails(req);
     const isBusinessRegistered = presenterAccountDetails?.isBusinessRegistered;
+    const businessName = isBusinessRegistered ? presenterAccountDetails?.businessName : null;
```
[the remark this answers is not in the document]
Comment (ch-skathirvel) [reply src/utils/session.ts:43]: Removed null

File: test/routers/handlers/enter_business_name/enter.business.name.unit.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    it("should translate page title to Welsh",  async () => {
+        session.setExtraData(
+            ContextKeys.PRESENTER_ACCOUNT_SESSION_KEY,
+            paDetailsWithIsBusinessRegisteredFalse
+        );
+
+        const response = await request(app).get(PrefixedUrls.ENTER_BUSINESS_NAME + "?lang=cy");
+
+        expect(response.text).toContain(`<title>${ScreenFieldsWelsh.ENTER_BUSINESS_NAME_TITLE}</title>`);
+    });
```
[the remark this answers is not in the document]
Comment (ch-skathirvel) [reply test/routers/handlers/enter_business_name/enter.business.name.unit.ts:201]: Added test case with true

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by amartin7663 — APPROVED** [review]

**Pull request merged**